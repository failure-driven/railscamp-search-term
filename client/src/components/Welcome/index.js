import React from 'react';
import { Link } from 'react-router-dom';

export default () => (
  <div>
    <h2>Welcome to the Game</h2>
    <Link to="/rounds/1" data-start-button>Start Game</Link>
  </div>
)