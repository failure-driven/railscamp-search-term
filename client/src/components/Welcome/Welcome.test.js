import React from 'react';
import Welcome from './';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import { Link } from 'react-router-dom';

it('renders welcome message', () => {
  const wrapper = shallow(<Welcome />);
  const welcome = <h2>Welcome to the Game</h2>;
  expect(wrapper.contains(welcome)).toEqual(true);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

describe('button', () => {
  it('takes you to round 1 on click', () => {
    const wrapper = shallow(<Welcome />);
    expect(wrapper.find(Link).prop('to')).toEqual('/rounds/1');
  });
})