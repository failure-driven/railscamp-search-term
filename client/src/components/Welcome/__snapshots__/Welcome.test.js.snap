// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders welcome message 1`] = `
<div>
  <h2>
    Welcome to the Game
  </h2>
  <Link
    data-start-button={true}
    replace={false}
    to="/rounds/1"
  >
    Start Game
  </Link>
</div>
`;
